optaplanner-integration/optaplanner-spring-boot-starter/pom.xml

```diff
@@ -31,6 +31,17 @@
       <groupId>org.optaplanner</groupId>
       <artifactId>optaplanner-core</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.optaplanner</groupId>
+      <artifactId>optaplanner-test</artifactId>
+      <exclusions>
+        <!-- Do not leak junit into the production classpath -->
+        <exclusion>
+          <groupId>org.junit.jupiter</groupId>
+          <artifactId>junit-jupiter-api</artifactId>
+        </exclusion>
+      </exclusions>
```

Comment:
This approach seems to be the lesser evil. Wdyt?

Reply:
See my review comment. I don't understand why we're doing this on 7.x at all.

Reply:
Agreed that this should wait for 8.x. What do you think of the exclusion itself?

Reply:
-1. `optaplanner-test` ends up in `BOOT-INF/lib/` of the executable fat JAR built by `spring-boot-maven-plugin`.

Reply:
I fear so, indeed. Not sure if that's worth the benefit of making it easy to write optaplanner tests indeed. We need to discuss.

Reply:
Not worth it in my opinion. Adding a test scoped `optaplanner-test` dependency manually is not a big deal.

Reply:
@yurloc This is a very good point that need to be resolved before we can continue. Would you mind marking your review as "request changes" to signal that this PR needs work before being merged.